sycl/include/CL/sycl/marray.hpp

```diff
@@ -0,0 +1,355 @@
+//==----------------- marray.hpp --- SYCL types ----------------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// Implements marray classes.
+
+#pragma once
+
+#include <CL/sycl/aliases.hpp>
+#include <CL/sycl/detail/common.hpp>
+#include <CL/sycl/detail/generic_type_traits.hpp>
+#include <CL/sycl/detail/type_traits.hpp>
+#include <CL/sycl/half_type.hpp>
+
+#include <array>
+#include <cmath>
+#include <cstring>
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+
+/// Provides a cross-patform math array class template that works on
+/// SYCL devices as well as in host C++ code.
+///
+/// \ingroup sycl_api
+template <typename Type, int NumElements> class marray {
+  using DataT = Type;
+
+public:
+  using value_type = Type;
+  using reference = Type &;
+  using const_reference = const Type &;
+  using iterator = Type *;
+  using const_iterator = const Type *;
+
+private:
+  value_type m_Data[NumElements];
```

Comment:
For the most part of the code the convention is to call class members `MData`. Can this one also be updated?

Reply:
It was initially made consistent with `vec` implementation.  Made it `MData` now.